src/flex/agentApp/src/WatsonAssistantAgentAppPlugin.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the MIT License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://opensource.org/licenses/MIT
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+import * as Flex from '@twilio/flex-ui';
+import { FlexPlugin } from 'flex-plugin';
+
+const PLUGIN_NAME = 'WatsonAssistantAgentAppPlugin';
+
+export default class WatsonAssistantAgentAppPlugin extends FlexPlugin {
+  constructor() {
+    super(PLUGIN_NAME);
+  }
+
+  /**
+   * This code is run when your plugin is being started
+   * Use this to modify any UI components or attach to the actions framework
+   *
+   * @param flex { typeof Flex }
+   * @param manager { Flex.Manager }
+   */
+  init(flex: typeof Flex, manager: Flex.Manager) {
+    this.registerReducers(manager);
+
+    flex.CRMContainer.defaultProps.uriCallback = (task: any) => {
+      // Pull the session history from the task attributes
+      if (task?.attributes?.sessionHistoryKey) {
+        let { sessionHistoryKey } = task.attributes;
+        return `https://web-chat.global.assistant.watson.cloud.ibm.com/loadAgentAppFrame.html?session_history_key=${sessionHistoryKey}`;
+      }
+
+      return '';
```

Comment:
I noticed when doing a little work in web chat that we were defaulting to `''` for the session history key and I wanted to change it but decided against it since it could have been a breaking change. I think it's a good idea to avoid using `''` as a default unless a string really is what is needed. It can make it harder to figure out if the string is just wrong or if it's really "not present" or it can hide other problems if code happily uses the value instead of throwing an error. I would prefer to use `null` unless you have an explicit reason to need `''` as a general rule of thumb.

Reply:
I ended up adding a comment, we're required to return a `string`, the string returned typically has to point to an URL, if it's an empty string it wont' load anything and show the default Twilio CRM panel:
<img width="1792" alt="Screen Shot 2021-12-03 at 11 54 17 AM" src="https://user-images.githubusercontent.com/12901345/144641783-46d24dae-1311-4a9f-bc46-15ff021db64b.png">